### ISSUE TITLE
Refactor/rename directory structure

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#9940]
I'm thinking of ways to restructure our source code inside `lib/` for easier understanding. Ideas include:

- Moving `datasource`, `manager`, `platform` and `versioning` under a new `modules/` subdirectory
- Restructuring workers so that the source code is sorted closer to execution order

After these changes, it would leave the following under `lib/`: `config`, `constants`, `logger`, `types` and `util`.

**Comment by viceice** [comment]
This would help for docs too, because currently we need to change the levels, which should hopefully make cross links easier.

**Comment by rarkins** [comment]
I want to do this on the next major release:

```
lib/
  modules/
    datasource/
    manager/
    platform/
    versioning/
  workers/
    global/
      config/
      init/
      run/
      teardown/
    repository/
      config/
      extract/
      lookup/
      update/
        branch/
        pr/
```

From experimenting on modules today, I found that the follow-up tasks included:
- Updating the ts strict config
- Looking for `jest.mock('../` and adding an extra `../` to it
- Updating snapshots

**Comment by rarkins** [comment]
I know, but for some reason it's not getting all of them. It doesn't catch `jest.mock()` either

**Comment by PhilipAbed** [comment]
@viceice nope, github/ webstorm/ vs code, i checked all, none of them move imports or any hardcoded tests/mocks, doing everything manually with regexes

**Comment by rarkins** [comment]
I can confirm that `typescript.updateImportsOnFileMove.enabled` *does* work, especially for imports. But:
- it seems like VSCode can get overloaded and not catch them all, so I recommend small updates at a time (e.g. move one modules at a time)
- it misses `jest.mock()` because that's library-specific and not a JS/TS concept

**Comment by rarkins** [comment]
Let's do just the potentially "breaking" changes for now:
- All modules
- Move branch and pr workers under global worker

**Comment by rarkins** [comment]
Some observations:

VSCode indeed does a good job of updating imports. But beware:
- Make sure you've opened at least one .ts file since restarting, or else the "ESLint server" won't be running
- After moving a folder, and letting VSCode make the changes, you should wait until the errors/warnings in the bottom bar stabilize and no longer change in count. Only once this is stable can you be free to "Save All" tabs and then Close All

Most of what's left is `jest.import()` and a few `require()` statements which were missed

**Comment by rarkins** [comment]
This was mostly implemented in #14490. Remaining parts are no longer "breaking"
